client/components/LearnerViewDirective.js

```diff
@@ -543,21 +546,36 @@ tie.directive('learnerView', [function() {
           }, displaySeconds * SECONDS_TO_MILLISECONDS);
         };
 
-        var activateAutosaving = function() {
-          $interval(function() {
-            var currentQuestionId =
-              $scope.questionIds[$scope.currentQuestionIndex];
-            triggerAutosaveNotification(DISPLAY_AUTOSAVE_TEXT_SECONDS);
-            CodeStorageService.storeCode(
-              currentQuestionId, $scope.code, language);
-          }, DEFAULT_AUTOSAVE_SECONDS * SECONDS_TO_MILLISECONDS);
+        $scope.activateAutosaving = function() {
+          if (!$scope.autosaveOn) {
+            $scope.autosaveOn = true;
+            autosaveCancelPromise = $interval(function() {
+              var currentQuestionId =
+                $scope.questionIds[$scope.currentQuestionIndex];
+              var storedCode = CodeStorageService.loadStoredCode(
```

Comment:
@rabidbit I can do that. But let me do some more tests about that since it's not a one line change. I actually need to update storedCode once currentQuestionId changes. (e.g. user navigates to next question)